central/views/imagecveflat/view_impl.go

```diff
@@ -0,0 +1,200 @@
+package imagecveflat
+
+import (
+	"context"
+	"sort"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/views"
+	"github.com/stackrox/rox/central/views/common"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/contextutil"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/postgres"
+	"github.com/stackrox/rox/pkg/postgres/walker"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/search"
+	pgSearch "github.com/stackrox/rox/pkg/search/postgres"
+	"github.com/stackrox/rox/pkg/search/postgres/aggregatefunc"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+var (
+	queryTimeout = env.PostgresVMStatementTimeout.DurationSetting()
+
+	log = logging.LoggerForModule()
+)
+
+type imageCVEFlatViewImpl struct {
+	schema *walker.Schema
+	db     postgres.DB
+}
+
+func (v *imageCVEFlatViewImpl) Count(ctx context.Context, q *v1.Query) (int, error) {
+	if err := common.ValidateQuery(q); err != nil {
+		return 0, err
+	}
+
+	var err error
+	q, err = common.WithSACFilter(ctx, resources.Image, q)
+	if err != nil {
+		return 0, err
+	}
+
+	queryCtx, cancel := contextutil.ContextWithTimeoutIfNotExists(ctx, queryTimeout)
+	defer cancel()
+
+	var results []*imageCVEFlatCount
+	results, err = pgSearch.RunSelectRequestForSchema[imageCVEFlatCount](queryCtx, v.db, v.schema, common.WithCountQuery(q, search.CVE))
+	if err != nil {
+		return 0, err
+	}
+	if len(results) == 0 {
+		return 0, nil
+	}
+	if len(results) > 1 {
+		err = errors.Errorf("Retrieved multiple rows when only one row is expected for count query %q", q.String())
+		utils.Should(err)
+		return 0, err
+	}
+	return results[0].CVECount, nil
+}
+
+func (v *imageCVEFlatViewImpl) Get(ctx context.Context, q *v1.Query, options views.ReadOptions) ([]CveFlat, error) {
+	if err := common.ValidateQuery(q); err != nil {
+		return nil, err
+	}
+
+	var err error
+	// Avoid changing the passed query
+	cloned := q.CloneVT()
+	cloned, err = common.WithSACFilter(ctx, resources.Image, cloned)
+	if err != nil {
+		return nil, err
+	}
+
+	var cveIDsToFilter []string
+	// Skip the pre-filter for the flattened model as performance should be improved.
+	if !features.FlattenCVEData.Enabled() && (cloned.GetPagination().GetLimit() > 0 || cloned.GetPagination().GetOffset() > 0) {
```

Comment:
If I understand correctly, all of this is only meant to run when FlattenCVEData is enabled, correct? If so, then you don't need to get the CVE IDs first.